src/pairwise_alltoallv.h

```diff
@@ -1,4 +1,124 @@
+#ifndef PAIRWISE_ALLTOALLV_H
+#define PAIRWISE_ALLTOALLV_H
+
 #include <mpi.h>
 
-int Pairwise_Alltoallv(const void *sendbuf, const int *sendcounts, const int *sdispls, MPI_Datatype sendtype,
-                       void *recvbuf, const int *recvcounts, const int *rdispls, MPI_Datatype recvtype, MPI_Comm comm);
+/*
+  MPI_Alltoallv which can handle large counts.
+  
+  Template parameter T should be an integer type large enough to store the largest
+  count on any MPI rank.
+*/
+template<typename T>
+int Pairwise_Alltoallv(const void *sendbuf, const T *sendcounts, const T *sdispls, MPI_Datatype sendtype,
```

Comment:
Out of general curiosity, why did you move the definition from `pairwise_alltoallv.cpp` to `pairwise_alltoallv.h`? Is there an advantage to one over the other?

Reply:
I changed it into a template function because I wanted the option of making the counts an 8 byte integer type. It's used with integer counts elsewhere (which should probably be fixed, but maybe not in this branch).

Reply:
I think templates have to be put in header files, so the function body has to go in pairwise_alltoallv.h now.